packages/cohort/src/replicator2/service.rs

```diff
@@ -0,0 +1,167 @@
+            tokio::select! {
+                is_decision = replicator.receive() => {
+                    if !is_decision {
+                        continue
+                    }
+                    metric.start();
+
+                    let mut recent_version = 0;
+                    loop {
+                        let next = replicator.get_next_statemap();
+                        if next.is_none() {
+                            break;
+                        }
+
+                        m_f_pick.clock_start();
+                        let (statemap, decided_at) = next
+                            .map(|(statemap, version, decided_at)| (StateMapWithVersion { statemap, version }, decided_at))
+                            .unwrap();
+                        m_f_pick.clock_end();
+
+                        let v = statemap.version;
+                        if recent_version == statemap.version {
+                            log::warn!(" will not schedule the same statemap ver({})", v);
+                            break;
+                        } else {
+                            recent_version = statemap.version;
+                        }
+
+                        m_f_send.clock_start();
+                        let rslt_send = tx.send((statemap, decided_at.unwrap_or(0))).await.map_err(|e| format!("Error: {}", e));
+                        m_f_send.clock_end();
+
+                        if let Err(e) = rslt_send {
+                            log::warn!("Unable to send statemap to installer. {}", e);
+                        } else {
+                            log::debug!(" scheduled statemap to install ver({})", v);
+                        }
+                    }
+
+                    let elapsed = metric.clock_end();
+                    if let Some(frequency) = metrics_print_frequency {
+                        if elapsed >= frequency {
+                            metric.sample_end();
+                            log::warn!(
+                                "METRIC (replicator) : {}, | lookups: {}, | sends: {}, cap: {}",
+                                metric.print("s"),
+                                m_f_pick.print("ms"),
+                                m_f_send.print("ms"),
+                                tx.capacity(),
+                            );
+
+                            // replicator.commit().await;
+                        }
+                    }
+                }
+
+                opt_install_resp = rx_install_response.recv() => {
+                    if let Some(InstallOutcome::Success { version, started_at: _started_at, finished_at: _finished_at }) = opt_install_resp {
+                        // ack += 1;
+                        if let Err(e) = replicator.update_suffix(version) {
+                            log::warn!("Error updating suffix for version: {}. Error: {}", version, e);
+                        } else {
+                           // let _ = replicator.receiver.commit(version).await;
```

Comment:
We will need to look at the `commit` strategy. One way to do would be similar to that done on `certifier` to maintain `tpl (topic partition list)` and at set interval commit using `tpl`

Reply:
https://jira.kindredgroup.com/browse/KDT-67